test/cmockatest.c

```diff
@@ -352,6 +353,62 @@ RTR_TEST_END
 RTR_TEST_START(dirfd)
 RTR_TEST_END
 
+/* netdb functions test */
+RTR_TEST_START(gethostbyname)
+	rtr_gethostbyname("www.google.com");
+RTR_TEST_END
+
+RTR_TEST_START(gethostbyaddr)
+	struct in_addr addr;
+
+	inet_pton(AF_INET, "64.233.189.147", &addr);
+	rtr_gethostbyaddr(&addr, sizeof(addr), AF_INET);
+RTR_TEST_END
+
+RTR_TEST_START(sethostent)
+	rtr_sethostent(1);
+RTR_TEST_END
+
+RTR_TEST_START(endhostent)
+	rtr_endhostent();
+RTR_TEST_END
+
+RTR_TEST_START(gethostent)
+	struct hostent *hent;
+	hent = rtr_gethostent();
+RTR_TEST_END
+
+RTR_TEST_START(gethostbyname2)
+	rtr_gethostbyname2("www.google.com", AF_INET);
+	rtr_gethostbyname2("www.google.com", AF_INET6);
+RTR_TEST_END
+
+RTR_TEST_START(getaddrinfo)
+	struct addrinfo hints, *result;
+
+	memset(&hints, 0, sizeof(struct addrinfo));
+	hints.ai_family = AF_UNSPEC;
+	hints.ai_socktype = SOCK_DGRAM;
+	hints.ai_flags = 0;
+	hints.ai_protocol = 0;
+
+	result = rtr_getaddrinfo("www.google.com", NULL, &hints, &result);
```

Comment:
You should use assert_* macros [1] here(and in similar cases) to check if result is appropriate. I.e.:

assert_int_equal(result, 0);

Btw, 'result' should probably be int, not struct addrinfo *.

https://api.cmocka.org/group__cmocka__asserts.html